kyaml/kio/kio.go

```diff
@@ -99,7 +104,7 @@ func (p Pipeline) ExecuteWithCallback(callback PipelineExecuteCallbackFunc) erro
 		}
 		result = append(result, nodes...)
 	}
-	if len(result) == 0 {
+	if len(result) == 0 && !p.ContinueIfInputEmpty {
```

Comment:
I prefer to remove this if statement altogether.

Reply:
Although no test fails if we delete this `if`, however I am worrying about this will affects some user's behavior because pipeline is so primary in kyaml. @monopole What do you think?

Reply:
Agree with @prachirp , and I share your worry.

In favor of dropping the if.  All functions should be able to deal with empty lists.

Keep the test.  Maybe add another with some Filters but no Inputs (or Inputs that do nothing), to provide regression.